### PR TITLE
Use kapp to manage deploys from source

### PR DESCRIPTION
**Pull request by scothis**
`kapp` is a Carvel tool that manages k8s resources. Like kubectl, kapp
is a client side cli that operates on vanilla k8s yaml.

kapp will:
- track resources it creates
- provide a summary of the changes it is about to apply
- optionally, diff the current and new state
- optionally, prompt to confirm the changeset
- watch resource for them to become ready. (i.e. Deployment rollouts are
  complete and new Pods are ready)
- prune fields no longer set (unlike `kubectl apply` which merges)
- delete unused resources (unlike `kubectl apply` which orphens)

Interally, kapp uses ConfigMaps to track which resources it is
responsible for managing. These ConfigMaps need to be created in a
namespace. We default this to kube-system, but can be overridden by
setting KAPP_NS. Additional options can be set by overridding KAPP_ARGS.

https://carvel.dev/kapp/

Signed-off-by: Scott Andrews <scott@andrews.me>

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -65,14 +66,18 @@ run: manifests generate fmt vet ## Run a controller from your host.
...
+KAPP_NS ?= kube-system
+
 install: manifests ## Install CRDs into the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/crd | $(KO) apply -f -
+	$(KAPP) deploy -a service-binding -n $(KAPP_NS) $(KAPP_ARGS) -f <($(KUSTOMIZE) build config/crd)
 
-uninstall: manifests ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/crd | $(KO) delete -f -
+uninstall: ## Uninstall CRDs from the K8s cluster specified in ~/.kube/config.
+	$(KAPP) delete -a service-binding -n $(KAPP_NS) $(KAPP_ARGS)
```
Comment (pedjak) [inline Makefile:77]: shouldn't this line be:

```
$(KAPP) delete -a service-binding -n $(KAPP_NS) $(KAPP_ARGS) -f <($(KUSTOMIZE) build config/crd)
```

File: Makefile [hunk 2]
```diff
@@ -65,14 +66,18 @@ run: manifests generate fmt vet ## Run a controller from your host.
...
 
 deploy: manifests ## Deploy controller to the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/default | $(KO) apply -f -
+	$(KAPP) deploy -a service-binding -n $(KAPP_NS) $(KAPP_ARGS) -f <($(KUSTOMIZE) build config/default | $(KO) resolve $(KO_ARGS) -f -)
 
 undeploy: ## Undeploy controller from the K8s cluster specified in ~/.kube/config.
-	$(KUSTOMIZE) build config/default | $(KO) delete -f -
+	$(KAPP) delete -a service-binding -n $(KAPP_NS) $(KAPP_ARGS)
```
Comment (pedjak) [inline Makefile:83]: similar question as above

File: Makefile (hunk 1 above)
Comment (scothis) [reply Makefile:77]: `kapp delete` does not accept files. It will delete all of the resources associated with the app.

**Review by pedjak — APPROVED** [review]

**Comment by scothis** [comment]
Superseded by #119

**Pull request closed without merging**